### DISCUSSION TITLE
tell mediacms to re-encode videos

### DISCUSSION THREAD

**Discussion by mashwino** [mediacms-io/mediacms#416] [Q&A] [answered]
Hello,
first of all thank you for your work, mediacms is amazing :-)
By mistake I removed an encoding profle from my mediacms and the encoded videos which was using it went away, is there a way to tell mediacms to re-encode all the original videos ?

Thank you.

**Comment by mgogoulos** [comment, marked as answer]
Hi, 

you cannot do it through the web but you can do it with some help from the console...
If you've used the single server installation, enter the Django shell

```
# cd /home/mediacms.io/mediacms
# source  /home/mediacms.io/bin/activate
# python manage.py shell
```
inside the shell do 

```
from files.models import *
```

see list of active EncodeProfiles

```
EncodeProfile.objects.filter(active=True)
<QuerySet [<EncodeProfile: h264-240>, <EncodeProfile: h264-360>, <EncodeProfile: h264-480>, <EncodeProfile: h264-720>, <EncodeProfile: h264-1080>, <EncodeProfile: vp9-1080>, <EncodeProfile: preview>]>
```

select the one that interests you. In my case it will be the second one
```
profile = EncodeProfile.objects.filter(active=True)[1]
profile
<EncodeProfile: h264-360>
```

now encode all videos through it. This will complete in a few seconds but start background jobs for the encoding work. 
```
for video in Media.objects.filter(media_type='video'):
    video.encode(profiles=[profile])
```

Let me know if this helps

**Reply by mashwino** [reply]
Exactly what I needed, thanks a lot.

**Reply by JustinBack** [reply]
If you are looking for a script, I've wrote one you can place in the root of mediaCMS:

```py
import os
os.environ.setdefault("DJANGO_SETTINGS_MODULE", "cms.settings")
import django
django.setup()
from files.models import *

print(EncodeProfile.objects.filter(active=True))

profile_name = input("Enter Encode Profile Name: ")
token = input("Enter Friendly Media Token: ")


profile = EncodeProfile.objects.get(name=profile_name)
video = Media.objects.get(friendly_token=token)
video.encode(profiles=[profile])

print("Media Encoding queued!")
```

Would love to see web implementation though.

**Reply by mgogoulos** [reply]
that could be an option for the single media page, and/or the manage media pages to perform it on more media at once...
